src/clj_bugsnag/core.clj

```diff
@@ -71,26 +72,42 @@
         (recur collected next)
         collected))))
 
+(def unrolled-exception-cache (cache/lru-cache-factory {}))
+
+(defn exception-cache-key [ex project-ns include-src?] (hash [ex project-ns include-src?]))
+
+(defn- unroll-cached [ex project-ns include-src?]
+  (let [k (exception-cache-key ex project-ns include-src?)]
+    (get (cache/through-cache unrolled-exception-cache
+                              k
+                              (fn [_] (unroll ex project-ns include-src?)))
+         k)))
+
 (defn exception->json
   [exception {:keys [api-key project-ns context group group-fn user
-                     severity version environment meta include-src?]
-              :or   {api-key      (env :bugsnag-key)
-                     project-ns   "\000"
-                     severity     "error"
-                     version      @git-rev
-                     environment  "production"
-                     include-src? true}
+                     severity version environment meta include-src? use-exception-cache?]
+              :or   {api-key              (env :bugsnag-key)
+                     project-ns           "\000"
+                     severity             "error"
+                     version              @git-rev
+                     environment          "production"
+                     include-src?         true
+                     use-exception-cache? true}
               :as   options}]
-  (let [ex        (parse-exception exception)
-        base-meta (if-let [d (ex-data exception)]
-                    {"ex–data" d}
-                    {})]
+  (let [ex         (parse-exception exception)
+        cached-ex? (cache/has? unrolled-exception-cache (exception-cache-key ex project-ns include-src?))
+        base-meta  (merge (if-let [d (ex-data exception)]
+                            {"ex–data" d}
+                            {})
+                          {"cached-exception?" cached-ex?})]
```

Comment:
I decided to this in for now because it seems like it could be useful, but I'm not sure really.

Reply:
Suggesting `(merge {cached? ...} (when-let [] {exdata ...}))`